src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const _state = { ...state };
+  const states = [];
+
+  for (const action in actions) {
+    switch (actions[action].type) {
```

Comment:
Can't you just write like this?
```suggestion
    switch (action.type) {
```